lib/utils/makeRequest.js

```diff
@@ -52,15 +52,15 @@ makeRequest.prototype.open = function() {
 
     var request = xmlHttpRequest();
 
-    if(this.password) {
-        request.open('POST', this.provider, true, this.tokenOrUsername, this.password);
-    } else {
-        request.open('POST', this.provider, true);
-    }
+
+    request.open('POST', this.provider, true);
+
     request.setRequestHeader('Content-Type','application/json');
     request.setRequestHeader('X-IOTA-API-Version', '1');
 
-    if (!this.password && this.tokenOrUsername) {
+    if (this.password) {
+        request.setRequestHeader('Authorization', 'Basic ' + Buffer.from(params.user + ':' + params.password).toString('base64'));
```

Comment:
`params` are `undefined` at this point. I believe this should be `this.tokenOrUsername` & `this.password`

Reply:
Solved in https://github.com/iotaledger/iota.js/pull/382/commits/b9585c9b0e2d50d88ec79f7e3b266118011c1e58